subprojects/enterprise-operations/src/main/java/org/gradle/execution/work/WorkExecutionBuildOperationType.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.execution.work;
+
+import org.gradle.internal.operations.BuildOperationType;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Captures the execution of a unit of work. This includes all operations related to Gradle's bookkeeping and optimizations.
+ * Among other things it includes snapshotting inputs and outputs, up-to-date checks, build cache operations.
+ * It also wraps the {@link WorkActionExecutionBuildOperationType execution of the action}.
+ *
+ * The operations can be nested, i.e. when a task executes a worker API action.
+ *
+ * @see WorkActionExecutionBuildOperationType
+ */
+public interface WorkExecutionBuildOperationType extends BuildOperationType<WorkExecutionBuildOperationType.Details, WorkExecutionBuildOperationType.Result> {
+
+    interface Details {
+
+        /**
+         * The type of work represented by the build operation.
+         */
+        Class<?> getWorkType();
+
+        /**
+         * Path to the owning build.
+         */
+        String getBuildPath();
+
+        /**
+         * A textual identifier for the unit of work when available.
+         *
+         * Tasks will expose their task paths here. Artifact transforms and workers don't expose anything here.
+         */
+        Optional<String> getIdentity();
```

Comment:
This is what I'm unsure about. Should we capture something for artifact transforms here? How about work actions?